### PR TITLE
Add docs for IIS Compression

### PR DESCRIPTION
**Pull request by bangbingsyb**
Previews:

[IIS Compression Overview](https://review.docs.microsoft.com/en-us/iis/extensions/iis-compression/iis-compression-overview?branch=user%2Fyashi%2Fiiscompression-docs)

[Using IIS Compression](https://review.docs.microsoft.com/en-us/iis/extensions/iis-compression/using-iis-compression?branch=user%2Fyashi%2Fiiscompression-docs)

### CODE REVIEW

File: iis/extensions/iis-compression/iis-compression-overview.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+> The article provides an overview of IIS Compression.
+
+## What is IIS Compression?
+
+**IIS Compression** is a bundle of two IIS compression scheme providers **iisbrotli.dll** and **iiszlib.dll** that support IIS to compress HTTP response contents.
+**iisbrotli.dll** supports **Brotli** compression, while **iiszlib.dll** supports both **Gzip** and **Deflate** compression.
+
+#### What is a compression scheme provider?
+
+An IIS compression scheme provider is a pluggable extension of IIS **StaticCompressionModule** and **DynamicCompressionModule** and provides algorithm for one or multiple HTTP compression schemes.
```
[the remark this answers is not in the document]
Comment (shirhatti) [reply iis/extensions/iis-compression/iis-compression-overview.md]: I'm going to replace algorithm with implementation. I feel like that reads better.

File: iis/extensions/iis-compression/iis-compression-overview.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Brotli compression is a relatively new addition to the HTTP compression.
+The encoding algorithm and format of Brotli are specified in [Requests For Comment (RFC) specification 7932](https://www.ietf.org/rfc/rfc7932.txt), and response data streams compressed with Brotli have the content encoding type "br".
+Brotli is capable of offering higher compression ratio than **Gzip** and **Deflate**, with the cost of lower compression speed.
+The Brotli decompression on the client (browser) side is in general faster.
+Therefore, enabling Brotli compression for the IIS web server using **iisbrotli.dll** can potentially bring more performance benefit especially for serving static contents.
+
+#### Better Gzip Compression Scheme Provider
+
+While **iiszlib.dll** included in **IIS Compression** offers same **Gzip** and **Deflate** compression as the default provider **gzip.dll**, **iiszlib.dll** is more up-to-date with slightly better performance in terms of CPU usage and better dynamic compression support owning to the zlib library.
```
[the remark this answers is not in the document]
Comment (shirhatti) [reply iis/extensions/iis-compression/iis-compression-overview.md]: s/owning/owing

File: iis/extensions/iis-compression/using-iis-compression.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+More details on how to configure rewrite rules can be found in
+[Creating Rewrite Rules for the URL Rewrite Module](https://docs.microsoft.com/en-us/iis/extensions/url-rewrite-module/creating-rewrite-rules-for-the-url-rewrite-module).
+
+## Testing IIS Compression
+
+Testing **IIS Compression** is as simple as opening a browser, requesting certain contents from the IIS server, and checking the requests and responses through the developer tools of the browser.
+
+[![](using-iis-compression/samples/test_compression.jpg)](using-iis-compression/samples/test_compression.jpg)
+
+> [!NOTE]
```
[the remark this answers is not in the document]
Comment (shirhatti) [reply iis/extensions/iis-compression/using-iis-compression.md]: I've removed the word note entirely. I don't think it adds much value here.

File: iis/extensions/iis-compression/using-iis-compression.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+> To test **IIS Compression** for static content compression:
+> - Ensure the MIME type of the requested resource is enabled in the `<staticTypes>` collection in the `<httpCompression>` element.
+> - Ensure the requested resource size is larger than `minFileSizeForComp` specified in `<httpCompression>` element.
+> - Ensure the "hit frequency" threshold is reached for the requested URL. The threshold is specified by the `frequentHitThreshold` and `frequentHitTimePeriod` attributes in the `<serverRuntime>` element. Alternatively,  set the value of the `staticCompressionIgnoreHitFrequency` attribute in the `<httpCompression>` element as `true` to disable the "hit frequency" check just for testing purpose.
+
+> [!NOTE]
```
[the remark this answers is not in the document]
Comment (shirhatti) [reply iis/extensions/iis-compression/using-iis-compression.md]: Likewise. I've removed the word note here as well

**Comment by shirhatti** [comment]
@Rick-Anderson I've updated this doc per your suggestions. Can you have another look?

**Comment by Rick-Anderson** [comment]
@shirhatti  Looks great. You can check my two commits, pure style guide changes, all minor. Next time I'll PR the PR and do all the work for you.

**Comment by shirhatti** [comment]
Thanks Rick! I'm going to rebase and merge

File: iis/extensions/iis-compression/iis-compression-overview.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+# IIS Compression Overview
+
+By [Yanbing Shi](https://github.com/bangbingsyb)
+
+This article provides an overview of IIS Compression.
+
+## What is IIS Compression
+
+**IIS Compression** is a bundle of two IIS compression scheme providers *iisbrotli.dll* and *iiszlib.dll* that support IIS to compress HTTP response contents.
```
Comment (jimmyca15) [inline iis/extensions/iis-compression/iis-compression-overview.md]: The grammar of this sentence is weird. 

Maybe change `that support IIS to compress HTTP response contents.` to `that expand IIS's capability to compress HTTP responses`

File: iis/extensions/iis-compression/iis-compression-overview.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+*iisbrotli.dll* supports **Brotli** compression, while *iiszlib.dll* supports both **Gzip** and **Deflate** compression.
+
+#### What is a compression scheme provider
+
+An IIS compression scheme provider:
+
+- Is a pluggable extension of IIS `StaticCompressionModule` and `DynamicCompressionModule`.
+- Provides an implementation for one or multiple HTTP compression schemes.
+
+An IIS compression scheme provider is a pluggable extension of IIS `StaticCompressionModule` and `DynamicCompressionModule` and provides algorithm for one or multiple HTTP compression schemes.
```
Comment (jimmyca15) [inline iis/extensions/iis-compression/iis-compression-overview.md]: Isn't this just a repetition of the two bullets above?

File: iis/extensions/iis-compression/iis-compression-overview.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+IIS `StaticCompressionModule` and `DynamicCompressionModule`:
+
+- Load the registered compression scheme providers into worker process at runtime.
```
Comment (jimmyca15) [inline iis/extensions/iis-compression/iis-compression-overview.md]: into **a** worker process

Comment (jimmyca15) [reply iis/extensions/iis-compression/iis-compression-overview.md]: or into worker process**es**

**Comment by shirhatti** [comment]
@jimmyca15 Whoops. I already merged. I'll create a new PR to address your feedback

**Comment by Rick-Anderson** [comment]
Or let @jimmyca15  Edit in browser and create a PR.

**Comment by jimmyca15** [comment]
@shirhatti @Rick-Anderson whichever one is fine for me. I can do the PR if you want Sourabh.

**Comment by Rick-Anderson** [comment]
@jimmyca15  if you could do the PR I can follow up.

**Pull request merged**